### PR TITLE
Allow routes to a service in warning status

### PR DESCRIPTION
**Pull request by erikvanoosten**
Currently Fabio only creates routes to services in the passing state. We have a use case for also wanting to use services in warning status. In consul the warning status indicates that the service is still good to go, but requires attention to keep it that way.

In this PR the acceptable statuses can be configured in fabio.properties where the default is just "passing" for backward compatibility.

As a bonus a few minor typos were fixed.

### CODE REVIEW

File: registry/consul/passing_test.go [hunk 1]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/hashicorp/consul/api"
 )
 
+//Add Warning, Unkown
```
Comment (magiconair) [inline registry/consul/passing_test.go]: Pls drop

Comment (erikvanoosten) [inline registry/consul/passing_test.go]: That's already gone in the last commit. I couldn't figure out how to squash the commits. I can spend more time on that if you wish.

File: config/config.go [hunk 2]
```diff
@@ -70,14 +70,15 @@ type File struct {
...
-	CheckTimeout  time.Duration
+	Addr             string
+	Token            string
+	KVPath           string
+	TagPrefix        string
+	Register         bool
+	ServiceAddr      string
+	ServiceName      string
+	ServiceTags      []string
+	AcceptableStates []string
```
Comment (magiconair) [inline config/config.go]: s/AcceptableStates/States/

File: fabio.properties [hunk 3]
```diff
@@ -186,6 +186,10 @@
 #
 # registry.consul.kvpath = /fabio/config
 
+# The comma separated list of statuses that are acceptable for using a service.
+# Possible values are: `passing`, `warning`, `critical` and `unknown`.
```
Comment (magiconair) [inline fabio.properties]: Pls replace backticks with single quotes

File: registry/consul/passing.go [hunk 4]
```diff
@@ -19,7 +19,7 @@ func passingServices(checks []*api.HealthCheck) []*api.HealthCheck {
 		}
 
 		// then make sure the service health check is passing
```
Comment (magiconair) [inline registry/consul/passing.go]: Update comment

**Comment by erikvanoosten** [comment]
> s/AcceptableStates/States/

In the last commit the field was renamed to `AcceptableStatus` to reflect Consul terminology. I have no idea what the plural is of _status_ is.

File: fabio.properties (hunk 3 above)
Comment (erikvanoosten) [inline fabio.properties]: Already done in the last commit.

**Comment by erikvanoosten** [comment]
I am closing this and I will create a new PR.

**Pull request closed without merging**